recipes/llvm-core/all/conanfile.py

```diff
@@ -1,429 +1,512 @@
+                "zstd::libzstd_static": "zstd::zstd",
+                "LibXml2::LibXml2": "libxml2::libxml2",
+                "ZLIB::ZLIB": "zlib::zlib"
+            }
+            for dep in deps_list.split(";"):
+                match = match_genex.search(dep)
+                if match:
+                    yield match.group(1)
+                else:
+                    replacement = replacements.get(dep)
+                    if replacement:
+                        yield replacement
+                    elif dep.startswith("-l"):
+                        yield dep[2:]
+                    else:
+                        yield dep
+
+        def _parse_deps(deps_list):
+            data = {
+                "requires": [],
+                "system_libs": []
+            }
+            windows_system_libs = [
+                "ole32",
+                "delayimp",
+                "shell32",
+                "advapi32",
+                "-delayload:shell32.dll",
+                "uuid",
+                "psapi",
+                "-delayload:ole32.dll"
+            ]
+            for component in _sanitized_components(deps_list):
+                if component in windows_system_libs:
                     continue
-                elif dep.startswith('LLVM'):
-                    components[dep]
-                elif dep in cmake_targets:
-                    dep = cmake_targets[dep]
-                elif os.path.exists(dep):
-                    dep = os.path.splitext(os.path.basename(dep))[0]
-                    dep = dep.replace('lib', '')
-                dep = dep.replace('-l', '')
-
-                if dep in dummy_targets.keys():
-                    components[lib].extend(dummy_targets[dep])
-                    components[lib] = list(set(components[lib]))
+                if component in ["rt", "m", "dl", "pthread"]:
+                    data["system_libs"].append(component)
                 else:
-                    components[lib].append(dep)
-
-            alias_targets = {}
-            old_alias_targets = {}
-            for component, _ in components.items():
-                alias_targets[component] = "LLVM::{}".format(component)
-                old_alias_targets["llvm-core::{}".format(component[4:].replace('LLVM', '').lower())] = "LLVM::{}".format(component)
-
-            # TODO: to remove in conan v2 once cmake_find_package_* generators removed
-            self._create_cmake_module_alias_targets(
-                os.path.join(self.package_folder, self._alias_module_file_rel_path),
-                alias_targets
-            )
+                    data["requires"].append(component)
+            return data
 
-            self._create_cmake_module_alias_targets(
-                os.path.join(self.package_folder, self._old_alias_module_file_rel_path),
-                old_alias_targets
-            )
+        cmake_exports = self.load(Path(self.package_folder) / "lib" / "cmake" / "llvm" / "LLVMExports.cmake")
```

Comment:
```suggestion
        cmake_exports = Path(self.package_folder, "lib", "cmake", "llvm", "LLVMExports.cmake").read_text("utf8")
```
`load(self, ...)` does not convert `\r\n` to `\n` on Windows, which can cause the regular expressions to fail unexpectedly.

I was bitten by this when applying a similar approach in the VTK recipe: https://github.com/conan-io/conan-center-index/blob/26a09d5b9c589206551669e75ebec365dff6368c/recipes/vtk/all/conanfile.py#L849-L853